hathor/util.py

```diff
@@ -215,7 +215,12 @@ def json_loadb(raw: bytes) -> Dict:
 
     # XXX: from Python3.6 onwards, json.loads can take bytes
     #      See: https://docs.python.org/3/library/json.html#json.loads
-    return json.loads(raw)
+    try:
+        return json.loads(raw)
+    except UnicodeDecodeError as exc:
+        # We cannot do `doc=raw` because it expects a str and there
+        # is no way to decode it.
+        raise json.JSONDecodeError(msg=str(exc), doc='', pos=exc.start) from exc
```

Comment:
We could do `doc=raw.hex()`, if that would help identify an error it's better than nothing at all.

Reply:
Nice idea. Thanks!